### PR TITLE
add Termination

### LINKED ISSUE #2266: ebiten: API to terminate an application explicitly
Now an Ebiten application can be terminated by returning an user-defined error at Update, but this is a little counterintuitive: even though the application exits in a regular way, the program treats 'error' values.

Instead of treating errors, what about having an explicit API to terminate an application, like `ebiten.Terminate`?

### PR DESCRIPTION
**Pull request by superloach**
should fix #2266

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -167,6 +171,9 @@ func (i *imageDumperGame) Layout(outsideWidth, outsideHeight int) (screenWidth,
 	return i.game.Layout(outsideWidth, outsideHeight)
 }
 
+// RegularTermination is a special error which indicates Game termination without error.
+var RegularTermination = ui.RegularTermination
```
Comment (hajimehoshi) [inline run.go:175]: Let's use `Termination`

Comment (superloach) [reply run.go:175]: yes, I was about to push the change :)

**Comment by superloach** [comment]
time to grab a coffee while tests run :coffee:

**Comment by hajimehoshi** [comment]
Isn't the base branch a little outdated? Now the `RunGame` comment should indicate how to terminate the game

**Comment by superloach** [comment]
I still haven't figured out how to rebase stuff properly in git, but I'll attempt it lol

**Comment by superloach** [comment]
eugh. git moment

**Pull request closed without merging**